src/registry/cache.rs

```diff
@@ -35,11 +38,18 @@ impl LocalRegistry {
 
     /// "Downloads" a package from the local filesystem
     pub async fn download(&self, dependency: Dependency) -> miette::Result<Package> {
+        let DependencyManifest::Remote(ref manifest) = dependency.manifest else {
+            return Err(miette!(
+                "unable to serialize version of local dependency ({})",
+                dependency.package
+            ));
+        };
```

Comment:
Nit: not useful code, `dependency_version_string` already does the check

Reply:
Yes but that is not technically part of the contract that `dependency_version_string` communicates – ie. although this is the current behavior the check for local dependencies is implicitly happening, rather than explicitly